### ISSUE TITLE
mbpfan does not detect the applesmc plugin 

### ISSUE THREAD

**Issue by Dizspy** [linux-on-mac/mbpfan#259]
I have Mac mini 2018  (Macmini8,1)

I've installed proxmox with custom linux kernel  v5.19.17-1 for T2 support
https://github.com/AdityaGarg8/pve-edge-kernel-t2

applesmc plugin successfully loaded
`root@proxmox:/# lsmod |grep smc`
`applesmc               32768  0`

sensors are successfully detecting all the values:
`root@proxmox:/# sensors`
`applesmc-acpi-0`
`Adapter: ACPI interface`
`fan1:        2530 RPM  (min = 1700 RPM, max = 4922 RPM)`
`TA0V:         +33.8°C`
`TC0F:         +84.0°C`
`TC0P:         +56.8°C`
`TC1C:         +69.0°C`
`TC2C:         +68.0°C`
`TC3C:         +66.0°C`
`TC4C:         +70.0°C`
`TC5C:         +66.0°C`
`TC6C:         +64.0°C`
`TCGC:         +62.0°C`
`TCSA:         +66.0°C`
`TCXC:         +69.5°C`
`TCaP:         +58.5°C`
`TH0F:         +51.8°C`
`TH0a:         +51.8°C`
`TH0b:         +50.2°C`
`TIED:         +61.5°C`
`TM0P:         +56.2°C`
`TM0V:         +62.8°C`
`TPCD:         +73.0°C`
`TPSD:         +55.2°C`
`TS0V:         +49.0°C`
`TTTD:         +70.8°C`
`TTXD:         +75.5°C`
`TW0P:         +56.5°C`
`TW1P:         +58.8°C`
`TW2P:         +59.5°C`
`TaLC:         +54.0°C`
`TaRC:         +60.0°C`

Anyway, mbpfan does not detect the applesmc plugin
`root@proxmox:/# mbpfan`
`mbpfan needs applesmc support. Please either load it or build it into the kernel. Exiting.`

How can I manually point mbpfan to detect the applesmc correctly to make it work?

**Comment by Redecorating** [comment]
Upstream mbpfan doesn't yet have the right folder path to work with the patched T2 kernels (#236 ), please have a look at this https://wiki.t2linux.org/guides/fan/ for now.

**Comment by zft9xgy** [comment]
Hi, @Dizspy some time ago I checked this issue and I had the same problem, it seems that mbpfan could not read the applesmc and only took the coretemp temperature, which was lower than what my applesmc was reading.

I have developed a script that may solve your problem, as it also solved mine [https://github.com/zft9xgy/mac-temp2fan-controler](https://github.com/zft9xgy/mac-temp2fan-controler) the code is still in its first version so it is possible to find some bugs.

**Comment by astropeak** [comment]
> Hi, [@Dizspy](https://github.com/Dizspy) some time ago I checked this issue and I had the same problem, it seems that mbpfan could not read the applesmc and only took the coretemp temperature, which was lower than what my applesmc was reading.
> 
> I have developed a script that may solve your problem, as it also solved mine https://github.com/zft9xgy/mac-temp2fan-controler the code is still in its first version so it is possible to find some bugs.

This works, thank!

**Comment by gaul** [comment]
I'm happy to merge PRs to improve fan or sensor support -- please test and submit them since I lack hardware for these.

**Comment by zft9xgy** [comment]
Yes, that is probably the best idea at that time but I don't got enough skill on dev and got at that time so I choose to build my own...

And right now I don't got the hardware available, maybe in the future I will got it again and I will do the fork+pr. 

Thanks !!
